### PR TITLE
Developed auth-related features (such as signup, signin, signout etc)

### PR DESCRIPTION
**Pull request by kyuhyunhan**
코드리뷰 부탁드립니다.

### CODE REVIEW

File: src/common/guards/jwt-auth.guard.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    if (isPublic) {
+      return true;
+    }
+
+    return super.canActivate(context);
+  }
+
+  handleRequest<TUser>(_: any, user: TUser): TUser {
+    if (!user) {
+      throw new UnauthorizedException('🙅🏻 접근 권한이 없습니다.');
```
Comment (sbp-lab) [inline src/common/guards/jwt-auth.guard.ts]: 👍🎉

File: src/user/constants/jwt.constant.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+export const ACCESS_TOKEN_STRATEGY = 'access-token';
+
+// 🚨 구현 생략
+export const REFRESH_TOKEN_STRATEGY = 'refresh-token';
+
+// 🚨 이 정보는 환경변수로 관리해야하는 정보입니다.
```
Comment (sbp-lab) [inline src/user/constants/jwt.constant.ts]: If you can, can you put comments in English from next time?

File: src/user/controllers/user.controller.ts [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+  @Get('/:user_id')
+  @HttpCode(HttpStatus.OK)
+  async findUserById(@Param('user_id') id: string) {
+    const user = await this.userService.findOneByAttr(
+      'user_id',
+      parseInt(id, 10),
+    );
+
+    if (user === null) {
+      throw new NotFoundException('👻 유저가 존재하지 않습니다.');
```
Comment (sbp-lab) [inline src/user/controllers/user.controller.ts]: Let's discuss if we put those kind of error message/alerts in English or Korean

File: src/user/services/auth.service.ts [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+@Injectable()
+export class AuthService {
+  constructor(
+    private readonly userService: UserService,
+    private readonly passwordService: PasswordService,
+    private readonly tokenService: TokenService,
+  ) {}
+
+  async validate(email: string, password: string) {
+    const user = await this.userService.findOneByAttr('email', email);
```
Comment (sbp-lab) [inline src/user/services/auth.service.ts]: For readability, type safety and maintainability, I recommend splitting this method into `findOneByEmail` and `findOneById`.

File: src/user/services/user.service.ts [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+@Injectable()
+export class UserService {
+  constructor(
+    @InjectRepository(User)
+    private readonly userRepository: Repository<User>,
+  ) {}
+
+  async findOneByAttr(attr: string, value: any): Promise<User | null> {
+    return await this.userRepository.findOne({ where: { [attr]: value } });
```
Comment (sbp-lab) [inline src/user/services/user.service.ts]: FYI: you can use findOneBy
Refer to https://g.co/gemini/share/5ff9a7be9445

example: 
// Simple match by ID
const user = await userRepository.findOneBy({ id: 123 });

**Review by indiegolab — COMMENTED** [review]
Since I'm not familiar with Auth, JWT, custom decorators and pipes I couldn't check these parts out, but other than that it looks good. 👍🏻

File: src/user/services/user.service.ts (hunk 5 above)
Comment (casysj) [reply src/user/services/user.service.ts]: +same

File: src/user/services/auth.service.ts (hunk 4 above)
Comment (casysj) [reply src/user/services/auth.service.ts]: findone이나 findoneby 쓰면 된다고 생각하고 있었는데 email, id용으로 따로 만드는게 좋은가요?

File: src/user/controllers/user.controller.ts [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+  @HttpCode(HttpStatus.OK)
+  async signin(@Body(ValidationPipe) body: SigninUsernDto): Promise<Tokens> {
+    const { email, password } = body;
+
+    return await this.authService.signin(
+      await this.authService.validate(email, password),
+    );
+  }
+
+  @Post('signout')
```
Comment (casysj) [inline src/user/controllers/user.controller.ts]: /signout하고
signout하고 동일하게 작동하나요?

File: src/user/constants/jwt.constant.ts (hunk 2 above)
Comment (kyuhyunhan) [reply src/user/constants/jwt.constant.ts]: I wrote everything in English again. Could you please check them?

File: src/user/controllers/user.controller.ts (hunk 3 above)
Comment (kyuhyunhan) [reply src/user/controllers/user.controller.ts]: I wrote everything in English again first (cuz thought It is better for the default) Could you please check them?

File: src/user/services/auth.service.ts (hunk 4 above)
Comment (kyuhyunhan) [reply src/user/services/auth.service.ts]: Thanks! I switched those methods' name into names you recommended. Please check them out☺️

Comment (kyuhyunhan) [reply src/user/services/auth.service.ts]: > findone이나 findoneby 쓰면 된다고 생각하고 있었는데 email, id용으로 따로 만드는게 좋은가요?

I thinks these cases depends on the team, context etc. In this case I kinda agreed to @indiegolab's to make the name more explicit.

File: src/user/services/user.service.ts (hunk 5 above)
Comment (kyuhyunhan) [reply src/user/services/user.service.ts]: Thanks guys! I changed them all. please check them again.

File: src/user/controllers/user.controller.ts (hunk 6 above)
Comment (kyuhyunhan) [reply src/user/controllers/user.controller.ts]: Yes sure I tested out again. But I changed it to '/signout' for the unity of codes. Thanks for pointing it out.

File: src/user/services/auth.service.ts (hunk 4 above)
Comment (sbp-lab) [reply src/user/services/auth.service.ts]: > findone이나 findoneby 쓰면 된다고 생각하고 있었는데 email, id용으로 따로 만드는게 좋은가요?

We can use `findOneBy` of repository and create separate services by using that.

**Review by indiegolab — APPROVED** [review]

**Review by casysj — APPROVED** [review]
lgfm

**Pull request merged**